CMakeLists.txt

```diff
@@ -78,14 +78,26 @@ SET(support_SOURCES
 )
 ADD_LIBRARY(support STATIC ${support_SOURCES})
 
+SET(optimizer_SOURCES
+  src/emscripten-optimizer/parser.cpp
+  src/emscripten-optimizer/simple_ast.cpp
+  src/emscripten-optimizer/optimizer-shared.cpp
+)
+ADD_LIBRARY(optimizer STATIC ${optimizer_SOURCES})
+
+
+SET(pass_SOURCES
+   src/pass.cpp
+   src/passes/Print.cpp
+ )
+ADD_LIBRARY(pass STATIC ${pass_SOURCES})
```

Comment:
This is an odd way to organize things. In general I'd like static libs to contain everything under a folder, and nothing else (i.e. folder hierarchy reflects library hierarchy). It makes it way easier to understand how things work.

pass.cpp should move under passes/, and parres/ contains a bunch of passes which should be in passes/.

The "right" way to keep this clean is probably to use cmake's add_subdirectory from the root cmake file, and then have each subdir create whatever it wants (static libraries most likely).